### PR TITLE
Checking for nans and rejecting models that evaluate to nan

### PR DESCRIPTION
**Pull request by clegaspi**
Latest commit should do it...haven't run unit tests yet or removed my unneeded comments

### CODE REVIEW

**Comment by mkhorton** [comment]
There are a lot of commented-out code in this commit, could you delete the old code first and then I'll take a look?

**Comment by clegaspi** [comment]
Right, I committed in hurry to go home last night. I'll clear it out. Was considering moving the code to check into a helper function within the class, but not sure if it's useful to do that.

**Comment by mkhorton** [comment]
Ok, this also needs a test + then can be merged

**Comment by mkhorton** [comment]
(An appropriate place for a test would be in test_models.py, with a new method like test_model_returns_nan() or similar)

File: propnet/core/models.py [hunk 1]
```diff
@@ -194,13 +193,32 @@ def evaluate(self, symbol_quantity_dict, allow_failure=True):
...
         provenance = ProvenanceElement(
             model=self.name, inputs=list(symbol_quantity_dict.values()))
         out = self.map_symbols_to_properties(out)
         for symbol, value in out.items():
             try:
-                out[symbol] = Quantity(symbol, value, self.unit_map.get(symbol),
+                quantity = Quantity(symbol, value, self.unit_map.get(symbol),
                                        provenance=provenance)
+
+                if quantity.symbol.category in ('property', 'condition'):
```
Comment (montoyjh) [inline propnet/core/models.py]: This is generally very solid; you might want to put these lines (i. e. 203-212 into a function or method (perhaps bound to Quantity) just to use less real estate in Model.evaluate.

**Comment by clegaspi** [comment]
At Joey's suggestion, moved nan checking to Quantity class function. Wrote appropriate tests for that function and to check model failure when nans present. Left comments about how we treat complex values.

I think it's important we scrub nan values to make sure they don't propagate through the graph traversal. However, the source of the nan values was a PyModel function which probably should have returned a complex value rather than nan. I'm looking into why it does, but not getting too distracted from more important things. I think we need to make sure propnet does a good job sanitizing bad output before we have public contributors of models.

**Comment by mkhorton** [comment]
Thanks 👍

**Pull request merged**